### PR TITLE
Mercurial support (ensure clean status, version bump, tag, push)

### PR DESCRIPTION
**Pull request by sjrmanning**
First of all thanks for all of your work on this project, it is amazing and a total revelation! I've started bringing it in at my workplace and setting it up for all of our projects.

My company is using mercurial for source control, so there were a few custom actions I had to implement to mimic the support fastlane has for git. I'm not sure if this is much use to anyone else as mercurial isn't too widely used, but I thought I'd share these back just in case.

All credit and my thanks to @lmirosevic for his git actions, which these were largely based off (especially the `hg_commit_version_bump` action). Cheers!

### CODE REVIEW

File: docs/Actions.md [hunk 1]
```diff
@@ -729,6 +730,58 @@ reset_git_repo(
 
 [MindNode](https://github.com/fastlane/examples/blob/master/MindNode/Fastfile) uses this action to reset temporary changes of the project configuration after successfully building it.
 
+## Using mercurial
+
+### hg_ensure_clean_status
+Along the same lines as the [`ensure_git_status_clean`](#ensure_git_status_clean) action, this is a sanity check to ensure the working mercurial repo is clean.  Especially useful to put at the beginning of your Fastfile in the `before_all` block.
```
Comment (KrauseFx) [inline docs/Actions.md]: There is a double space after `clean.  `

File: lib/fastlane/actions/hg_add_tag.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        [
+          FastlaneCore::ConfigItem.new(key: :tag,
+                                       env_name: "FL_HG_TAG_TAG",
+                                       description: "Tag to create")
+        ]
+      end
+
+      def self.author
+        # credits to lmirosevic for original git version
+        "sjrmanning"
```
Comment (KrauseFx) [inline lib/fastlane/actions/hg_add_tag.rb]: I guess the `author` method should soon be an array :smile: @lmirosevic @sjrmanning

**Comment by KrauseFx** [comment]
@sjrmanning thank you so much for your contribution, this is amazing! :+1: 

I have never used mercurial myself and have no real way to test it. Are you using those actions already? Any questions that came up? 

Would you be able to bring the code coverage change to +-0%? Right now it's a decrease of 0.64%

Thanks again, really appreciated :rocket:

**Comment by sjrmanning** [comment]
@KrauseFx no problem at all, my pleasure! 

I am currently using these actions in a Fastfile like [this here](https://gist.github.com/sjrmanning/5f0b221b768b263b481c).

The problem with the coverage is the big version bump commit action, same with the git version it's a little hard to test, but I've done my best. My ruby experience is very limited to be honest so I hope what I've done with that spec is okay (the action uses `Helper.is_test` extensively to skip over some things). It's mainly testing how it acts with changed files and what's expected to have changed during a version bump commit.

**Comment by KrauseFx** [comment]
Looks good, thanks @sjrmanning :+1: Test coverage is now -0.14%, which is okay. Thanks again for your contribution, this will be in the next release :rocket:

**Comment by sjrmanning** [comment]
Great! Thanks @KrauseFx! :smile:

**Pull request merged**